CMIS Connector

Nirthika opened the pull request:
## Purpose
> EI Connector for CMIS (Content Management Interoperability Services)

## Goals
> Write initial version of EI Connector for CMIS

## Approach
> N/A

## User stories
> N/A

## Release note
> N/A

## Documentation
> Git Doc added

## Training
> N/A

## Certification
> N/A

## Marketing
> N/A

## Automation tests
 - Unit tests 
   > N/A
 - Integration tests
   > Four test cases

## Security checks
 > Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
Ran FindSecurityBugs plugin and verified report? yes
Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
> N/A

## Related PRs
> N/A

## Migrations (if applicable)
> N/A

## Test environment
- Maven 3.x
- Java 1.8
Platform:
 - UBUNTU 16.04
 - WSO2 EI 6.1.1
 
## Learning
> About CMIS

Nirthika committed:
CMIS Connector

Initial commit

sivaramya commented on src/main/resources/config/component.xml:
```diff
@@ -0,0 +1,26 @@
+            <description>configure cmis connector</description>
+        </component>
+    </subComponents>
+</component>
```
.xml files should end with a new line. Please add file-ending new line.

Nirthika replied:
Could you please explain why we need to add new line .xml files?

sivaramya commented on src/main/assembly/assemble-connector.xml:
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<assembly>
```
License header is missing

kalaiyarasiganeshalingam commented on src/test/resources/automation.xml:
```diff
@@ -0,0 +1,266 @@
+            </extentionClasses>
+        </platformExecutionManager>
+
+
```
remove the extra spaces

Nirthika committed:
Fix

Update README file, add license and add blank line at end of files

Nirthika committed:
Update files

Change discribtion for cmisBlockingInvocation, Hardcode the mendatory parameter which are not CMIS vender specific and remove two unwanted test cases.

hasuniea commented on src/main/resources/config/init.xml:
```diff
@@ -0,0 +1,33 @@
+    <parameter name="cmisUsername" description="Username to log in the Alfresco server"/>
+    <parameter name="cmisPassword" description="Password to log in the Alfresco server"/>
+    <parameter name="cmisInstanceUrl" description="URL of the Alfresco CMIS instance"/>
+    <parameter name="cmisBlockingInvocation" description="If it is true the same thread will serve the response otherwise
```
Does above line exceed the character length

Nirthika committed:
Fix

Format one line

dimslee reviewed (changes requested):
Please follow the proper standards for commenting.

http://chris.beams.io/posts/git-commit/

You cannot answer "No" to security section of the GitHub template.

Nirthika committed:
Update doc

Remove cmisSelector parameters
